nixpkgs/nixpkgs.bzl

```diff
@@ -185,3 +186,63 @@ stderr:
   code=exec_result.return_code,
   stdout=exec_result.stdout,
   stderr=exec_result.stderr))
+
+def _nixpkgs_cc_autoconf_impl(repository_ctx):
+  # Calling repository_ctx.path() on anything but a regular file
+  # fails. So the roundabout way to do the same thing is to find
+  # a regular file we know is in the workspace (i.e. the WORKSPACE
+  # file itself) and then use dirname to get the path of the workspace
+  # root.
+  workspace_file_path = repository_ctx.path(
+    Label("@nixpkgs_cc_toolchain//:WORKSPACE")
+  )
+  workspace_root = repository_ctx.execute(
+    ["dirname", workspace_file_path]
+  ).stdout.rstrip()
+
+  # Make a list of all available tools in the Nix derivation. Override
+  # the Bazel autoconfiguration with the tools we found.
+  res = repository_ctx.execute(["ls", workspace_root + "/bin"])
+  if res.return_code == 0:
+    overriden_tools = {
+      tool: repository_ctx.path(Label("@nixpkgs_cc_toolchain//:bin/" + tool))
+      for tool in res.stdout.splitlines()
+    }
+    cc_autoconf_impl(repository_ctx, overriden_tools = overriden_tools)
+  else:
+    fail("Cannot list content of @nixpkgs_cc_toolchain:" + res.stderr)
+
+_nixpkgs_cc_autoconf = repository_rule(
+  implementation = _nixpkgs_cc_autoconf_impl
+)
+
+def nixpkgs_cc_configure(
+    repository = None,
+    repositories = None,
+    nix_file_content = """
+      with import <nixpkgs> {}; buildEnv {
+        name = "bazel-cc-toolchain";
+        paths = [ gcc binutils ];
+      }
+    """):
+  """Use a CC toolchain from Nixpkgs.
+  
+  By default, Bazel auto-configures a CC toolchain from commands (e.g.
+  `gcc`) available in the environment. To make builds more hermetic, use
+  this rule to specific explicitly which commands the toolchain should
+  use.
+  """
+  if repository and repositories or not repository and not repositories:
+    fail("Specify one of repository or repositories (but not both).")
+  elif repository:
+    repositories = {"nixpkgs": repository}
+
+  nixpkgs_package(
+    name = "nixpkgs_cc_toolchain",
+    repositories = repositories,
+    nix_file_content = nix_file_content,
+    build_file_content = """exports_files(glob(["bin/*"]))""",
+  )
+  _nixpkgs_cc_autoconf(name = "local_config_cc")
+  native.bind(name = "cc_toolchain", actual = "@local_config_cc//:toolchain")
```

Comment:
[Using `bind` is not recommended](https://docs.bazel.build/versions/master/be/workspace.html#bind). Afaik it’s not deprecated though.

Reply:
This code was blindly copied from `cc_autoconf`. I think we should do whatever they are doing.